### ISSUE TITLE
Improve CupertinoContextMenu drag behavior.

### ISSUE THREAD

**Issue by abdalmonem** [flutter/flutter#160509]
### Use case

CupertinoContextMenu has a major native distinguish behavior in the dragging process.

Current behavior:
1 - When user long press on item the context appear but there is no interacting.
2 - The User need to release his finger and repress the preview area to interact with the menu.

https://github.com/user-attachments/assets/1a3de1a5-1540-47e9-8617-f3c224995625

The expected behavior:
1 - When user long press on item the context menu should appear
2 - While user finger keep finger down and swap the menu should keep interacting scrolling.

https://github.com/user-attachments/assets/0413149c-771b-4c9b-b4a8-16f7b4a0ca51

In both videos i does not released my finger after popup appear, you can reproduce this misbehave by keep your finger pressed down until pop-up show and try to move your finger around.

Also the drag cancel time is not same as the native, but this is not our topic today , as i see in the source code, there is a constants holding timing for drag, speed .. etc , so it well be easy to make this calibration latter.

### Proposal

I guess we need somehow a way to track the global gesture position , and pass it from the child to the parent and keep synchronizing tracking while preview showing.
I really thinking about solve this issue, but it need more digging to collect more information's about the reflection of IOS accessibility preferences and there impact on the context menu.
If any one have any suggestions that could help, please share them  to discuss.

**Comment by maheshj01** [comment]
Hi @abdalmonem,  Thanks for filing the issue, I am able to reproduce the issue on the latest stable and the master channel.     CupertinoContextMenu is not interactible until the long press gesture is released.


Flutter | IOS Native
:-: | :-:
<video src='https://github.com/user-attachments/assets/558b39f2-9169-448b-871d-04fbb9e493a4' width=180/> | <video src='https://github.com/user-attachments/assets/b61fb4a9-e155-4e45-b43d-3f82fdf5c9d3' width=180/> 


<details>
<summary>flutter doctor -v</summary>




```console
[✓] Flutter (Channel stable, 3.27.0, on macOS 15.1.1 24B2091 darwin-arm64, locale en-US)
    • Flutter version 3.27.0 on channel stable at /Users/mahesh/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8495dee1fd (7 days ago), 2024-12-10 14:23:39 -0800
    • Engine revision 83bacfc525
    • Dart version 3.6.0
    • DevTools version 2.40.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-35, build-tools 34.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.94.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.100.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 15.1.1 24B2091
      darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 15.1.1 24B2091
      darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome
      128.0.6613.85

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```
[!] Flutter (Channel master, 3.28.0-1.0.pre.123, on macOS 15.1.1 24B2091 darwin-arm64, locale en-US)
    • Flutter version 3.28.0-1.0.pre.123 on channel master at /Users/mahesh/Development/flutter_master
    • Framework revision a547c7f050 (4 days ago), 2024-12-13 22:11:39 -0500
    • Engine revision 4e4a430bcd
    • Dart version 3.7.0 (build 3.7.0-243.0.dev)
    • DevTools version 2.41.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use
      "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-35, build-tools 34.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.94.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.100.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 15.1.1 24B2091
      darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 15.1.1 24B2091
      darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome
      128.0.6613.85

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by abdalmonem** [comment]
@maheshj01 I already stared to investigation on this [issue/feature] , can you assign it to me ?

**Comment by maheshj01** [comment]
@abdalmonem I have assigned this issue to you.

**Comment by abdalmonem** [comment]
Blocked by [https://github.com/flutter/flutter/issues/126151], without fix this we cannot doing much , unless we made a big changes to use `Overlay` instead of push another route into navigation stack, or more painful way by overriding both Navigator, and NavigatorState.

**Comment by dkwingsmt** [comment]
It seems to me that this issue is a duplicate of https://github.com/flutter/flutter/issues/117936. Closing this one.
